src/components/toolbar/toolbar.js

```diff
@@ -86,7 +86,23 @@ const Toolbar = ({
    * @returns {void}
    */
   const onClearFilter = ({ value }) => {
-    const updatedValue = (Array.isArray(value) && value) || [value];
+    const updatedValue = [];
+
+    if (typeof value === 'string') {
+      updatedValue.push(value);
+    }
+
+    if (Array.isArray(value)) {
+      updatedValue.push(
+        ...value.map(obj => {
+          if (typeof obj === 'string') {
+            return obj;
+          }
+          return obj.name;
+        })
+      );
+    }
+
     updatedValue.forEach(val => clearField(val));
   };
```

Comment:
future: refactor should include making some flavor of this "on clear" func dep injected for testing